gensim/test/test_fasttext.py

```diff
@@ -325,65 +325,122 @@ def test_doesnt_match(self):
             min_count=5, iter=5, batch_words=1000, word_ngrams=1, sample=1e-3, min_n=3, max_n=6,
             sorted_vocab=1, workers=1, min_alpha=0.0)
 
         lee_data = LineSentence(datapath('lee_background.cor'))
         model_gensim.build_vocab(lee_data)
-        orig0 = np.copy(model_gensim.wv.syn0[0])
-        model_gensim.train(lee_data, total_examples=model_gensim.corpus_count, epochs=model_gensim.iter)
-        self.assertFalse((orig0 == model_gensim.wv.syn0[0]).all())  # vector should vary after training
-        self.compare_with_wrapper(model_gensim, model_wrapper)
+        orig0 = np.copy(model_gensim.wv.vectors[0])
+        model_gensim.train(lee_data, total_examples=model_gensim.corpus_count, epochs=model_gensim.epochs)
+        self.assertFalse((orig0 == model_gensim.wv.vectors[0]).all())  # vector should vary after training
 
-    def test_sg_hs_against_wrapper(self):
-        if self.ft_path is None:
-            logger.info("FT_HOME env variable not set, skipping test")
-            return
+        sims_gensim = model_gensim.wv.most_similar('night', topn=10)
+        sims_gensim_words = (list(map(lambda x: x[0], sims_gensim)))  # get similar words
+        expected_sims_words = [
+            u'night,',
+            u'night.',
+            u'eight',
+            u'nine',
+            u'overnight',
+            u'crew',
+            u'overnight.',
+            u'manslaughter',
+            u'north',
+            u'flight']
+        overlap_count = len(set(sims_gensim_words).intersection(expected_sims_words))
+        self.assertGreaterEqual(overlap_count, 3)
+
+    def test_cbow_neg_training(self):
+
+        model_gensim = FT_gensim(size=50, sg=0, cbow_mean=1, alpha=0.05, window=5, hs=0, negative=5,
+            min_count=5, iter=5, batch_words=1000, word_ngrams=1, sample=1e-3, min_n=3, max_n=6,
+            sorted_vocab=1, workers=1, min_alpha=0.0)
 
-        tmpf = get_tmpfile('gensim_fasttext.tst')
-        model_wrapper = FT_wrapper.train(ft_path=self.ft_path, corpus_file=datapath('lee_background.cor'),
-            output_file=tmpf, model='skipgram', size=50, alpha=0.025, window=5, min_count=5, word_ngrams=1,
-            loss='hs', sample=1e-3, negative=0, iter=5, min_n=3, max_n=6, sorted_vocab=1, threads=12)
+        lee_data = LineSentence(datapath('lee_background.cor'))
+        model_gensim.build_vocab(lee_data)
+        orig0 = np.copy(model_gensim.wv.vectors[0])
+        model_gensim.train(lee_data, total_examples=model_gensim.corpus_count, epochs=model_gensim.epochs)
+        self.assertFalse((orig0 == model_gensim.wv.vectors[0]).all())  # vector should vary after training
 
-        model_gensim = FT_gensim(size=50, sg=1, cbow_mean=1, alpha=0.025, window=5, hs=1, negative=0,
+        sims_gensim = model_gensim.wv.most_similar('night', topn=10)
+        sims_gensim_words = (list(map(lambda x: x[0], sims_gensim)))  # get similar words
```

Comment:
`sims_gensim_words = [word for (word, distance) in sims_gensim]` (here and everywhere)